### ISSUE TITLE
Syntax error with older perls (0.001 - 0.003)

### ISSUE THREAD

**Issue by eserte** [team-at-cpan/OpenTracing#1]
With older perls (5.22.4 and older) module compilation fails:
```
#   Failed test 'OpenTracing.pm loaded ok'
#   at t/00-compile.t line 50.
#          got: '65280'
#     expected: '0'
Array found where operator expected at /home/cpansand/.cpan/build/2019061616/OpenTracing-0.003-_VzsaB/blib/lib/OpenTracing/Log.pm line 46, at end of line
	(Missing operator before ?)
syntax error at /home/cpansand/.cpan/build/2019061616/OpenTracing-0.003-_VzsaB/blib/lib/OpenTracing/Log.pm line 46, near "->@* "
Compilation failed in require at /home/cpansand/.cpan/build/2019061616/OpenTracing-0.003-_VzsaB/blib/lib/OpenTracing.pm line 44.
BEGIN failed--compilation aborted at /home/cpansand/.cpan/build/2019061616/OpenTracing-0.003-_VzsaB/blib/lib/OpenTracing.pm line 44.
Compilation failed in require at -e line 1.
... (etc) ...
```

**Comment by vanHoesel** [comment]
Yep, it is sad that this software is using post-dereference. Making it unsuitable to be used in legacy code.

**Comment by tm604** [comment]
I'll release a version with the minimum perl defined, thanks for the report.

@vanHoesel this module was released after perl 5.24, and postfix dereference has been part of core Perl for several major releases. I have no plans for backwards compatibility myself, but I'd accept patches/PRs for supporting older versions within reason - as long as they don't make the code significantly harder to maintain :smile:

**Comment by tm604** [comment]
Fixed in version 0.004, https://metacpan.org/release/TEAM/OpenTracing-0.004/view/lib/OpenTracing.pm
